core/src/main/java/org/elasticsearch/index/mapper/NumberFieldMapper.java

```diff
@@ -574,17 +591,26 @@ Query rangeQuery(String field, Object lowerTerm, Object upperTerm,
                 int l = Integer.MIN_VALUE;
                 int u = Integer.MAX_VALUE;
                 if (lowerTerm != null) {
-                    l = parse(lowerTerm);
-                    if (includeLower == false) {
+                    l = parse(lowerTerm, true);
+                    // if the lower bound is decimal:
+                    // - if the bound is positive then we increment it:
+                    //      if lowerTerm=1.5 then the (inclusive) bound becomes 2
+                    // - if the bound is negative then we leave it as is:
+                    //      if lowerTerm=-1.5 then the (inclusive) bound becomes -1 due to the call to longValue
+                    boolean lowerTermHasDecimalPart = hasDecimalPart(lowerTerm);
+                    if ((!lowerTermHasDecimalPart && includeLower == false) ||
+                        (lowerTermHasDecimalPart && signum(lowerTerm) > 0)) {
```

Comment:
can you indent one level more so that it is easier to figure out what is part of the if statement and what is part of the inner block?